torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -517,6 +517,14 @@ torch::lazy::hash_t XLAGraphExecutor::GetGraphHash(
           res_hash, torch::lazy::Hash(buffer_donor_index));
     }
   }
+  if (ShardingUtil::GetAutoSharding()) {
```

Comment:
let's drop this if statement, consider a case where we have
```
if (feature_a_enabled) {
  coll.hash = combine(coll.hash , True)
}
if (feature_b_enabled) {
  coll.hash = combine(coll.hash , True)
}
```
in this case there will be a hash collision since hash is the same for either `a` or `b` is enabled. Let's always hash the `autosharding` and `XLA_AUTO_SPMD_MESH `.

Reply:
I was including the flag name to the string for this... but then I realize that hashing True twice would be different from hashing True once. We can do eitehr way... yea let's drop and make it "always"

Reply:
right, I was not worried about autosharding. I was worried about we have another feature like
```
if(ShardingUtil::GetAutoSharding()) {
}
if(feature_b_enabled) {
}
```
if `feature_b` also has a boolean hashing as well, there might be conflicts.

I think we can either
1. include string name in the hash(which is what you did) and put this hash in a if condition
2. only hash the boolean value but unconditionally hashing it

I think we usually do 2(for example for `force_ltc_tensor` config)